server/lib/schema/index.ts

```diff
@@ -163,7 +163,41 @@ const schema = new GraphQLSchema({
       workspace: {
         type: workspaceType,
         args: { id: { type: GraphQLString } },
-        resolve: resolver(models.Workspace)
+        resolve: resolver(models.Workspace, {
+          after: async (result, args, ctx) => {
```

Comment:
When exactly will this hook run?

Reply:
Whenever the workspace is fetched. Maybe it makes more sense to validate it when it's created. The main problem is I don't really see how this bug arose in the first place, i.e., how a workspace could have been created without having the associated three blocks (question, answer, scratchpad). So here I opted for the solution that's guaranteed to eliminate the bug. You'll never be able to fetch a workspace that's missing one of these required blocks.

Reply:
Would be good to understand how the situation could arise in the first place. (What if the reason it arose can also cause other errors?)

Reply:
Yeah definitely. If you look the Workspace model's `afterCreate` hook, you'll see the creation of the associated blocks is very straightforward. Also, we've only seen this error once out of 1000+ workspace creations. So my best guess is that understanding how this happened will be very difficult. Having more robust logging on the backend would help w/ future debugging efforts if someone like this happens again.